runtime/src/accounts.rs

```diff
@@ -278,26 +275,49 @@ impl Accounts {
                             })
                             .unwrap_or_default();
 
-                        if account.executable() && bpf_loader_upgradeable::check_id(account.owner())
+                        if account.executable()
+                            && demote_program_write_locks
+                            && message.is_writable(i, demote_program_write_locks)
+                            && !is_upgradeable_loader_present(message)
```

Comment:
can we hoist `is_upgradeable_loader_present` outside this loop?

Reply:
Yes, ofc, thanks!